abdiff/core/utils.py

```diff
@@ -0,0 +1,38 @@
+"""abdiff.core.utils"""
+
+import json
+import os
+from pathlib import Path
+
+from slugify import slugify
+
+from abdiff.config import Config
+
+CONFIG = Config()
+
+
+def get_job_slug_and_working_directory(job_name: str) -> tuple[str, Path]:
+    """Create working directory for new job by slugifying job name."""
+    job_slug = slugify(job_name)
+    return job_slug, Path(CONFIG.data_directory) / job_slug
+
+
+def update_or_create_job_json(job_name: str, new_job_data: dict) -> dict:
```

Comment:
To anyone who works' on functionality that will update the job's `job.json`, my hope is that this utility function would be reusable for that purpose.